Fix missing dependencies, CLI convention and convert to pyproject.toml

pmhahn opened the pull request:
I have it a try today and immediately run into 3 ploblems: #20 #21 #22
The first 2 #20 and #21 I was able to fix.
Bonus: convert `setup.*` to `pyproject.toml`

Commit:
Convert setup.* to pyproject.toml

Changed license to https://spdx.org/licenses/BSD-3-Clause.html
Dropped classifier "License :: OSI Approved :: BSD License"

`requirements.txt` cannot be merged as it still is used by:
- .readthedocs.yml
- .devcontainer/Dockerfile
- .devcontainer/devcontainer.json
What is `snooty`?

Signed-off-by: Philipp Hahn <p.hahn@avm.de>

Commit:
Replace deprecated pkg_resources.resource_filename

Dependency was missing.

Replace with `importlib.resources` available since Python 3.7.

Signed-off-by: Philipp Hahn <p.hahn@avm.de>
Closes: #20

Commit:
Replace deprecated pkg_resources.parse_version

Dependency was missing.

Depend on `packaging` for version comparison.

Signed-off-by: Philipp Hahn <p.hahn@avm.de>
Closes: #20

Commit:
Require command

Require at least one argument for command:

```console
$ flametrace
Traceback (most recent call last):
  File "…/bin/flametrace", line 10, in <module>
    sys.exit(cli())
             ^^^^^
  File "…/lib/python3.11/site-packages/click/core.py", line 1161, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "…/lib/python3.11/site-packages/click/core.py", line 1082, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "…/lib/python3.11/site-packages/click/core.py", line 1443, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "…/lib/python3.11/site-packages/click/core.py", line 788, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "…/src/flametrace/main.py", line 112, in cli
    argv0 = os.path.basename(command[0])
                             ~~~~~~~^^^
IndexError: tuple index out of range
```

See <https://click.palletsprojects.com/en/stable/arguments/>.

Signed-off-by: Philipp Hahn <p.hahn@avm.de>
Closes: #21

leifwalsh reviewed (approved):
Oh gosh I didn't notice this. Thank you very much! Sorry it took so long to merge.

leifwalsh commented:
hmm, gotta fix CI too

Commit:
GitHub: Update supported Python version

Python 3.7-9 have reached their End-of-life years ago. Remove them.
Instead add 3.10-14.
See <https://devguide.python.org/versions>.

Signed-off-by: Philipp Matthias Hahn <pmhahn@pmhahn.de>

pmhahn commented:
> hmm, gotta fix CI too

See https://devguide.python.org/versions/: All versions of Python 3.7-9 mentioned in `.github/workflows/python-package.yml` have reached their End-of-life and are no longer supported by See https://github.com/actions/python-versions#support-policy

Commit:
ReadTheDocs: Update build configuration

Python 3.8 has reached its End-of-life a year ago. Switch to 3.14.
See <https://docs.readthedocs.com/platform/stable/config-file/v2.html>

Signed-off-by: Philipp Matthias Hahn <pmhahn@pmhahn.de>

Commit:
requirements: Drop snooty

There is no package available by such name on pypi.org. Drop it.

Signed-off-by: Philipp Matthias Hahn <pmhahn@pmhahn.de>

leifwalsh committed:
yeahhhhhhhml

leifwalsh committed:
Click compatibility

leifwalsh committed:
don't worry about stderr, run flake8 on 3.14

leifwalsh committed:
test result.output not stdout

leifwalsh committed:
fix datetime parsing

leifwalsh committed:
Update python versions in tox.ini

leifwalsh committed:
set docs language to english

leifwalsh committed:
test command success first and show output

leifwalsh committed:
Update Dockerfile

leifwalsh committed:
Update devcontainer.json

leifwalsh committed:
fix sigterm parsing

leifwalsh commented:
such a pile of yaks had moved in

leifwalsh commented:
Ok, I can't reproduce that failure in codespaces. I need to put this down for a bit and come back to it, but we're a lot closer.

leifwalsh committed:
handle clone3
